review/get_review_ids.py

```diff
@@ -0,0 +1,235 @@
+# pylint: disable=import-error
+'''
+The logic for actually grabbing review content is contained here for the
+Review XBlock. This works by having a copy of the actual course a learner
+is interacting with that is hidden from learners. This copied course hosts
+the content we will show as review so the review content displayed has a
+fresh state and can be set to ungraded so it doesn't affect the learners'
+grades. There are two ways review content can be grabbed:
+    On a per problem basis
+    On a unit basis (this would be single view that contains multiple problems)
+'''
+
+import json
+import logging
+import random
+from datetime import datetime
+
+import crum
+import pytz
+from courseware.models import StudentModule
+from enrollment.api import add_enrollment, get_enrollment, update_enrollment
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from xmodule.modulestore.django import modulestore
+
+from .configuration import ENROLLMENT_COURSE_MAPPING, REVIEW_COURSE_MAPPING, TEMPLATE_URL
+
+log = logging.getLogger(__name__)
+
+# TODO: Switch to using CourseLocators and/or CourseKeys everywhere
+
+
+def get_problems(num_desired, current_course):
+    '''
+    Looks through all the problems a learner has previously loaded and randomly
+    selects num_desired of them. Also checks if the learner had originally
+    answered it correctly or incorrectly and after how many attempts.
+
+    Parameters:
+        num_desired (int): the number of desired problems to show the learner
+        current_course (CourseLocator): The course the learner is currently in
+
+    Returns a list of num_desired tuples in the form (URL to display, correct, attempts)
+    '''
+    user = crum.get_current_user()
+
+    enroll_user_in_review_course(user, current_course)
```

Comment:
Nit: This function has side-effects (enrolling the learner and deleting their state).  It's unclear from the docstring or the function name that these side-effects happen.  One possibility is to move out the enrollment call outside of this function.

Reply:
I'm not sure what you mean by deleting their state. Is that the re-enrolling them if they happen to unenroll from the course or are you referring to how the review version of the course will be hidden from the dashboard?

I'm happy to fix up the docstring to make anything more clear, just want to verify which parts are not clear. I'm also not sure what you mean about moving the enrollment call outside of the function. As in put it in a __main__ or just at the top where it gets executed upon accessing the file?